### PR TITLE
Fix Microsoft.VisualBasic Ref API surface issues

### LINKED ISSUE #27972: MatchingRefApiCompat issues with Microsoft.VisualBasic.
See baseline file
https://github.com/dotnet/corefx/blob/master/src/Microsoft.VisualBasic/src/MatchingRefApiCompatBaseline.netstandard.txt.

We should do one of the following:
1. Expose the APIs publicly in the reference assembly
2. Make the APIs in the implementation internal
3. Add a comment to the baseline file why you cannot do either (1) or (2) and keep them baselined.

See PR https://github.com/dotnet/corefx/pull/27881/ for extra context.

### PR DESCRIPTION
**Pull request by OmarTawfik**
Fixes #27972 

I made the following changes:
* If the type is not exposed through the ref assembly, make it internal.
* Unless the type was used via reflection, keep it public, and add an exception.
* If the type is not sealed in the implementation but in the ref assembly, seal it in the implementation.

### CODE REVIEW

**Comment by OmarTawfik** [comment]
@dotnet-bot test Windows x86 Release Build please

**Comment by OmarTawfik** [comment]
@VSadov @weshaggard 
Can you please take a look?

File: src/Microsoft.VisualBasic/src/MatchingRefApiCompatBaseline.netstandard.txt [hunk 1]
```diff
@@ -1,16 +1,7 @@
 Compat issues with assembly Microsoft.VisualBasic:
-TypesMustExist : Type 'Microsoft.VisualBasic.ControlChars' does not exist in the implementation but it does exist in the contract.
-CannotSealType : Type 'Microsoft.VisualBasic.HideModuleNameAttribute' is sealed in the implementation but not sealed in the contract.
-TypesMustExist : Type 'Microsoft.VisualBasic.Interaction' does not exist in the implementation but it does exist in the contract.
-MembersMustExist : Member 'Microsoft.VisualBasic.Strings.Left(System.String, System.Int32)' does not exist in the implementation but it does exist in the contract.
+
+# Loaded via reflection, so it should not be added to the ref assembly:
```
Comment (weshaggard) [inline src/Microsoft.VisualBasic/src/MatchingRefApiCompatBaseline.netstandard.txt]: Just for clarity I would change the comment to "Loaded via reflection so needs to be public in the implementation but we don't want to expose them publicly in the ref assembly"

**Comment by OmarTawfik** [comment]
@dotnet-bot test this

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

**Comment by OmarTawfik** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by OmarTawfik** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by OmarTawfik** [comment]
@dotnet-bot test Windows x64 Debug Build

**Comment by OmarTawfik** [comment]
@dotnet-bot test Windows x64 Debug Build

**Comment by OmarTawfik** [comment]
@dotnet-bot test Linux x64 Release Build

**Comment by alexperovich** [comment]
This is a known helix issue caused by issues with a rollout. We rolled-back and things are functional again. Please re-run the affected builds.

**Comment by riarenas** [comment]
We had a Kusto outage that caused all those problems. You should be able to try again now.  dotnet/core-eng#3072

**Comment by OmarTawfik** [comment]
@dotnet-bot test this please

**Comment by OmarTawfik** [comment]
@dotnet-bot test OSX x64 Debug Build

**Comment by weshaggard** [comment]
@OmarTawfik OSX leg is backed up don't block on it. Your change shouldn't cause any issues there.

**Comment by OmarTawfik** [comment]
@weshaggard alright. I'll merge now. Thanks :)

**Pull request merged**